codegen/src/types/tests.rs

```diff
@@ -371,6 +371,43 @@ fn compact_fields() {
     )
 }
 
+#[test]
+fn compact_generic_parameter() {
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct S {
+        a: Option<<u128 as codec::HasCompact>::Type>,
+    }
```

Comment:
Is it worth adding tests to check that some of the other TypePath variants lead to the output you expect?